tools/maf_cleanup.py

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/python
+import optparse
+
+### start of parsing parameters
+parser = optparse.OptionParser()
+parser.add_option('-i', '--input-file', action = 'store', dest = 'inputfile', help = "file name of input maf")
+parser.add_option('-o', '--output-file', action= 'store', dest = "outputfile", default=False, help = "name of output maf")
+
+(options, args) = parser.parse_args()
+input_file = options.inputfile
+output_file = options.outputfile
+
+if not input_file:
+	print 'ERROR: missing input MAF name.'
+	parser.print_help()
+elif not output_file:
+	print 'ERROR: missing output MAF name.'
+	parser.print_help()
+
+### end of parsing parameters
+
+### start of searching for header row
+input_maf = open (input_file, "r")
+header_line = ""
+for line in input_maf:
+	values = line.split("\t")
+	if values[0].upper() == "HUGO_SYMBOL":
+		header_line = line
+
+headers = header_line.rstrip().split("\t")
+keep_col_index_arr = [] #Index of columns to keep
+input_maf.close()
+### end of searching for header row
+
+### start of filtering out unwanted columns
+count = 0
+exon_number_index = -1
+exon_index = -1
+intron_index = -1
+oncotator_index_arr = []
+for header_item in headers:
+	if header_item.upper() == "EXON_NUMBER":
+		exon_number_index = count
+	elif header_item.upper() == "EXON":
+		exon_index = count
+	elif header_item.upper() == "INTRON":
+		intron_index = count
+	elif header_item[0:len("ONCOTATOR")].upper() == "ONCOTATOR":
+		oncotator_index_arr.append(count)
+	else:
+		keep_col_index_arr.append(count)
+	count += 1
+
+print "INDEX OF COLUMNS TO DROP >>>>>>>>>>>>>>"
+print "INTRON: " + str(intron_index)
+print "EXON_NUMBER: " + str(exon_number_index)
+print "EXON: " + str(exon_index)
+print "ONCOTATOR: " + ' '.join(str(index) for index in oncotator_index_arr)
+print "DROPPING COLUMNS ...."
+### end of filtering out unwated columns
+
+### Start of reconstructing file
+input_maf = open (input_file, "r")
+output_maf = open (output_file, "w")
+
+# write out the original extra header lines (sequences samples, versions, etc.)
+for line in input_maf:
+	values = line.split("\t")
+	if values[0].upper() != "HUGO_SYMBOL":
+		output_maf.write(line)
+	else:
+		break
+
+# write out header line
+for keep_col_index in keep_col_index_arr:
+	if headers[keep_col_index].upper() == "HUGO_SYMBOL":
+		output_maf.write("Hugo_Symbol")
+	elif headers[keep_col_index].upper() == "START_POSITION":
+		output_maf.write("\tStart_Position")
+	elif headers[keep_col_index].upper() == "END_POSITION":
+		output_maf.write("\tEnd_Position")
+	else:
+		output_maf.write("\t" + headers[keep_col_index])
+output_maf.write("\n")
+
+# write out content
+for line in input_maf:
+	line = line.rstrip()
+	col_counter = 0
+	cols = line.split("\t")
+	for keep_col_index in keep_col_index_arr:
+		if col_counter != 0:
+			output_maf.write("\t")
+			output_maf.write(cols[keep_col_index])
+		else:
+			if cols[keep_col_index].upper() == "1-FEB":
```

Comment:
the replacement rules should be a static file that is loaded into a dictionary .. then the logic here could look if the key is in the dictionary and the replacement value could be retrieved.  It makes the rules easier to maintain too.

Reply:
fixed